lib/blas_quda.cu

```diff
@@ -134,89 +134,127 @@ namespace quda {
       }
     };
 
-    void axpbyz(double a, const ColorSpinorField &x, double b, const ColorSpinorField &y, ColorSpinorField &z)
+    void axpbyz(cvector<double> &a, cvector_ref<const ColorSpinorField> &x, cvector<double> &b,
+                cvector_ref<const ColorSpinorField> &y, cvector_ref<ColorSpinorField> &z)
     {
-      instantiate<axpbyz_, Blas, true>(a, b, 0.0, x, y, x, x, z);
+      check_size(a, x, b, y, z);
+      for (auto i = 0u; i < x.size(); i++)
+        instantiate<axpbyz_, Blas, true>(a[i], b[i], 0.0, x[i], y[i], x[i], x[i], z[i]);
     }
 
-    void axy(double a, const ColorSpinorField &x, ColorSpinorField &y)
+    void axy(const cvector<Complex> &a, cvector_ref<const ColorSpinorField> &x, cvector_ref<ColorSpinorField> &y)
```

Comment:
Similar to above, `cvector<Complex>` to `cvector<double>`?

Reply:
Intentional 😄